gpu/gpu.go

```diff
@@ -92,14 +131,27 @@ func initGPUHandles() {
 	}
 
 	slog.Info("Detecting GPU type")
-	cudaLibPaths := FindGPULibs(cudaMgmtName, cudaMgmtPatterns)
-	if len(cudaLibPaths) > 0 {
-		cuda := LoadCUDAMgmt(cudaLibPaths)
+	// Prefer libcudart.so if it's available
+	cudartLibPaths := FindGPULibs(cudartMgmtName, cudartMgmtPatterns)
+	if len(cudartLibPaths) > 0 {
+		cuda := LoadCUDAMgmt(cudartLibPaths)
 		if cuda != nil {
 			slog.Info("Nvidia GPU detected")
 			gpuHandles.cuda = cuda
+			gpuHandles.cuda.lib_t = LibCUDART
 			return
 		}
+	} else {
```

Comment:
The reason I did it this way is because we were trying to incorporate libnvidia-ml.so and libcudart.so under the unified "cuda" handle. If I want them to be separate, I think I would need to create a different configuration similar to the first entry I did with the gpu_info_tegra.c, etc build. I'll see what I can come up with to remove the `else`, but one way or another would likely involve adding another GPU handle type.

Side note: I did see your PR regarding the creation of the common LLM libraries folder. I've already successfully loaded the packaged libraries in another branch, and it wouldn't be much to change the code to support the ~/.ollama/lib filepath. I did notice that ROCM libraries aren't packaged with llama.cpp and the only libraries that get bundled are the cuda libraries.

Reply:
>  I did notice that ROCM libraries aren't packaged

Unfortunately the ROCm libraries are massive, so bundling them inside the executable like we do with cuda isn't viable.  We're still trying to figure out the optimal approach for radeon cards which is part of the reason we still haven't officially launched support for them, even though they generally work on main.  It's plausible we may shift to an approach of decoupled dependency downloads and lazy-load them if they weren't pre-loaded by the installer flow.  That PR you mention is an incremental step that will help if we do go that route.

Reply:
Yeah, I saw the comments you included in the `gen_linux.sh` file regarding how big the ROCm libraries are being the reason they aren't bundled in. 

With loading the packaged drivers though, I had a hard time resolving the "chicken and egg" scenario presented: needed the library loaded to check if the device is present and ready to go, and needed to know if the device is present or not to figure out which library to load. In the end, I just loaded them all and let them fail until one was successful and then set that one as the library. I'll see if I can sneak in support for the new PR to load the pre-packaged cuda libraries.